### DISCUSSION TITLE
Poll: Package for Validators

### DISCUSSION THREAD

**Discussion by carlosforero** [cohesivestack/valgo#8] [Polls]
This poll is regarding the package for validators.

Currently, Valgo has only one package, `github.com/cohesivestack/valgo`, so using the library functionality requires using the valgo package:

```go
import "github.com/cohesivestack/valgo"

valgo.Is(valgo.String("name").Not().Blank())
```

So, the README suggests using an alias to make the code more readable:

```go
import v "github.com/cohesivestack/valgo"

v.Is(v.String("name").Not().Blank())
```

But what if the validators are moved to the package `valgo/v`, so that the core functionality of Valgo will be kept in the `valgo` package? This could make the difference between a Validation session and the Validators more clear, and make it more concise to get external validator libraries for Valgo from the community. Consider the example:

```go
import "github.com/cohesivestack/valgo"
import "github.com/cohesivestack/valgo/v"

valgo.Is(v.String("name").Not().Blank())
```

Suppose someone in the community has created a net validator for Valgo:

```go
import "github.com/cohesivestack/valgo"
import "github.com/cohesivestack/valgo/v"
import "github.com/someone/netv"

valgo.Is(v.String("name").Valid())
     .Is(netv.HttpUrl("url").Valid())
```

So the question is: Would you prefer to keep the current setup with only the `valgo` package, or add a package for the validators, `valgo/v`, and keep the core functionality in `valgo`?

**Comment by bokunodev** [comment]
i agree to separate the validation rules into a separate package. but i'm against `valgo/v`, we need a more meaningful name. like `valgo/rule` or something.

**Reply by carlosforero** [reply]
I see your point. Just I've been thinking in something very short for code readability. However I'm not sure about `rule` since in this library we have the concept that a Validator has Rules:

```go
valgo.Is( valgo.String("x", "field").Not().Blank().EqualTo("value") );

// valgo.Is( ... ) -> Create a Validation Session
// valgo.String("x", "field") -> The Validator
// .Not().Blank().EqualTo("value") -> The Rules

```

So, `valgo/validator` is so long. Maybe `valgo/val` but that could be confusing.

**Reply by PaluMacil** [reply]
Right now anyone can shorten it as recommended in the readme. If you use something like rule, val, or v, you would likely conflict with variable names in some projects, so people would still do the same thing--just different people. Anyone who aliased before would stop and people who didn't might start. That situation seems to point to not changing the API.

I don't see any reason to split up the project into different packages either when it's a fairly simple API. Everything basically belongs in one bucket, so moving the bucket over a couple inches doesn't really change your organization method.
